pkg/aggregator/aggregator.go

```diff
@@ -479,9 +479,12 @@ func (agg *BufferedAggregator) run() {
 			agg.flush()
 			addFlushTime("MainFlushTime", int64(time.Since(start)))
 			aggregatorNumberOfFlush.Add(1)
-		case sample := <-agg.dogstatsdIn:
-			aggregatorDogstatsdMetricSample.Add(1)
-			agg.addSample(sample, timeNowNano())
+		case samples := <-agg.dogstatsdIn:
+			aggregatorDogstatsdMetricSample.Add(int64(len(samples)))
+			ts := timeNowNano()
+			for _, sample := range samples {
```

Comment:
Nit-pick but: could we move this to `addSample` to keep all the logic there and this function as simple as possible ?